skimage/color/colorconv.py

```diff
@@ -804,6 +804,60 @@ def rgb2grey(rgb):
     return rgb2gray(rgb)
 
 
+def gray2rgba(image, alpha=None):
+    """Create a RGBA representation of a gray-level image.
+
+    Parameters
+    ----------
+    image : array_like
+        Input image.
+    alpha : array_like, optional
+        Alpha channel of the output image. It may be a scalar or an
+        array that can be broadcast to ``image``. If not specified it is
+        set to the maximum limit correspondin to ``image`` data type.
+
+    Returns
+    -------
+    rgba : ndarray
+        RGBA image. A new dimension of length 4 is added to input
+        image shape.
+
+    Raises
+    ------
+    ValueError
+        If ``alpha`` is a scalar and is not in ``image`` data type
+        limits.
+    ValueError
+        If ``alpha`` is an array and have a different shape then
+        ``image``.
+    ValueError
+        If ``alpha`` is an array and have a different data type then
+        ``image``.
+
+    """
+
+    arr = np.asarray(image)
+
+    alpha_min, alpha_max = dtype_limits(arr, clip_negative=False)
+
+    if alpha is None:
+        alpha = alpha_max
+    elif np.isscalar(alpha):
+        if alpha_min > alpha or alpha > alpha_max:
+            warn("alpha = {} is not in image data type limits."
+                 .format(alpha))
+
+    if not np.can_cast(alpha, image.dtype):
+        warn("alpha can't be safely cast to image dtype {}"
+             .format(arr.dtype.name))
+
+    rgba = np.empty(arr.shape + (4, ), dtype=arr.dtype)
+    rgba[..., :3] = arr[..., np.newaxis]
+    rgba[..., 3] = alpha
```

Comment:
Do we want to convert alpha to the appropriate units here? Using the convert function

Reply:
i sticked to what you advised before: "junk in -> junk out"...
I am not against the conversion here, I was even more strict in the first release of this PR since it was raising an error if alpha is an array of different dtype then image....

Reply:
I'll have to check the truth table for np.can_cast

Reply:
This is the only thing I would like to bring to the attention of the 2nd reviewer.

Reply:
So, my take is that we shouldn't raise for out of range (e.g. `alpha=2.5` in a float image), but we *should* raise on under/overflow, such as `alpha=4096` for a uint8 image. What do you think about this suggestion @rfezzani?

Reply:
I agree with @jni.

Also, I think no explicit conversion needs to be done by us.

Reply:
@jni and @stefanv, this is already checked [here](https://github.com/scikit-image/scikit-image/pull/4424/files#diff-9f4dedd98e7cca606dec9f0b792c1d94R845) ;-)